lib/lib.go

```diff
@@ -408,6 +408,19 @@ func NewInstance(ctx context.Context, repoPath string, opts ...Option) (qri *Ins
 			// we have a connection
 			log.Debugf("using RPC address %s", cfg.RPC.Address)
 			inst.rpc = rpc.NewClient(conn)
+
+			// TODO(arqu): I blindly assume this means we have a working API connection
+			// till the HTTPClient is fleshed out
+			maAddr, err := ma.NewMultiaddr(cfg.API.Address)
+			if err != nil {
+				return nil, err
+			}
+			httpAddr, err := manet.ToNetAddr(maAddr)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
zomg I've wanted this function for a long time.

We should move these into the `NewHTTPClient` constructor though, and pass `cfg.API.Address` directly